File: Sqrt(x).cpp
class Solution {
public:
    int mySqrt(int x) 
    {
        if(x == 0 || x == 1)
        {
            return x;
        }
        
        long long start = 2;
        long long end = x;
        
        
        while(start <= end)
        {
            long long mid = (start + end) / 2;
            
            if(mid * mid == x)
            {
                return mid;
            }
            
            if(mid * mid < x)
            {
                start = mid + 1;
            }
            
            if(mid * mid > x)
            {
                end = mid - 1;
            }
            
        }
        
        return start - 1;
        
    }
};

File: Sqrt(x).py
class Solution:
    def mySqrt(self, x: int) -> int:
        low=0
        high=x
        while low<high:
            mid=(low+high)//2
            if mid*mid==x:
                return int(mid)
            elif mid*mid<x:
                low=mid+1
            else:
                high=mid-1
        if low*low>x:
            return low-1
        else:
            return low